bundles/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/internal/handler/NetworkOceanicThingHandler.java

```diff
@@ -62,7 +63,8 @@ public void initialize() {
         NetworkOceanicBindingConfiguration config = getConfigAs(NetworkOceanicBindingConfiguration.class);
 
         try {
-            socket = new Socket(config.ipAddress, config.portNumber);
+            this.socket = new Socket(config.ipAddress, config.portNumber);
+            Socket socket = this.socket;
```

Comment:
This would avoid risking null, so the check in line 68 would not be needed:
```suggestion
            Socket socket = new Socket(config.ipAddress, config.portNumber);
            this.socket = socket;
```

Reply:
trye, thanks